EchoBot.cs

```diff
@@ -0,0 +1,177 @@
+
+                    var parsedResult = AdaptiveCard.FromJson(card);
+                    var attachment = new MessagingExtensionAttachment
+                    {
+                        ContentType = AdaptiveCard.ContentType,
+                        Content = parsedResult.Card,
+                    };
+
+                    var result = new MessagingExtensionResult("list", "result", new[] { attachment });
+
+                    return new MessagingExtensionResponse(result);
+                }
+                else
+                {
+                    return new MessagingExtensionResponse
+                    {
+                        ComposeExtension = new MessagingExtensionResult
+                        {
+                            Type = "silentAuth"
+                        }
+                    };
+                }
+            }
+            return null;
+        }
+
+        private string GetPostTransformedPFTToken(String preTransformedPFTToken)
```

Comment:
Comment which defines 
1. about paramenters
2. return value
3. Exceptions thrown in any case
4. overall information of what this function is doing.